modular_skyrat/modules/modular_items/code/modular_glasses.dm

```diff
@@ -0,0 +1,297 @@
+  hud_trait = TRAIT_SECURITY_HUD
+  glass_colour_type = /datum/client_colour/glass_colour/red
+
+/obj/item/clothing/glasses/hud/aviator/security/attack_self(mob/living/user)
+  if(toggleable && !user.incapacitated())
+    on = !on
+    if(on)
+      flash_protect = initial(flash_protect)
+      to_chat(usr, span_notice("You switch \the [src] to flash protection mode."))
+      hud_type = initial(hud_type)
+      hud_trait = initial(hud_trait)
+      tint = initial(tint)
+    else
+      flash_protect = FLASH_PROTECTION_FLASH
+      to_chat(usr, span_notice("You switch the [src] to HUD mode."))
+      hud_type = null
+      hud_trait = null
+      tint = 1
+    update_icon()
+    playsound(src, activation_sound, 50, TRUE)
+    user.update_inv_glasses()
+    user.update_action_buttons()
+    user.update_sight()
+  ..()
+
+// Medical Aviators
+/obj/item/clothing/glasses/hud/aviator/health
+  name = "medical HUD aviators"
+  desc = "A heads-up display that scans the humanoids in view and provides accurate data about their health status. This HUD has been fitted inside of a pair of sunglasses."
+  icon_state = "aviator_med"
+  off_state = "aviator"
+  toggleable = TRUE
+  vision_correction = FALSE
+  flash_protect = FLASH_PROTECTION_NONE
+  hud_type = DATA_HUD_MEDICAL_ADVANCED
+  hud_trait = TRAIT_MEDICAL_HUD
+  glass_colour_type = /datum/client_colour/glass_colour/lightblue
+
+// (Normal) meson scanner Aviators
+/obj/item/clothing/glasses/hud/aviator/meson
+  name = "meson HUD aviators"
+  desc = "Used by engineering and mining staff to see basic structural and terrain layouts through walls, regardless of lighting conditions. This HUD has been fitted inside of a pair of sunglasses."
+  icon_state = "aviator_meson"
+  off_state = "aviator"
+  toggleable = TRUE
+  vision_correction = FALSE
+  flash_protect = FLASH_PROTECTION_NONE
+  clothing_traits = list(TRAIT_MADNESS_IMMUNE)
+  darkness_view = 2
+  vision_flags = SEE_TURFS
+  lighting_alpha = LIGHTING_PLANE_ALPHA_MOSTLY_VISIBLE
+  glass_colour_type = /datum/client_colour/glass_colour/lightgreen
+
+// diagnostic Aviators
+/obj/item/clothing/glasses/hud/aviator/diagnostic
+  name = "diagnostic HUD aviators"
+  desc = "A heads-up display capable of analyzing the integrity and status of robotics and exosuits. This HUD has been fitted inside of a pair of sunglasses."
+  icon_state = "aviator_diagnostic"
+  off_state = "aviator"
+  toggleable = TRUE
+  vision_correction = FALSE
+  flash_protect = FLASH_PROTECTION_NONE
+  hud_type = DATA_HUD_DIAGNOSTIC_BASIC
+  hud_trait = TRAIT_DIAGNOSTIC_HUD
+  glass_colour_type = /datum/client_colour/glass_colour/lightorange
+
+// Science Aviators
+/obj/item/clothing/glasses/hud/aviator/science
+  name = "science aviators"
+  desc = "A pair of tacky purple aviator sunglasses that allow the wearer to recognize various chemical compounds with only a glance."
+  icon_state = "aviator_sci"
+  off_state = "aviator"
+  toggleable = TRUE
+  vision_correction = FALSE
```

Comment:
```suggestion
```
vision_correction is already set to false on the parent obj, you don't need to set it again.